tests/storage/linstor/test_linstor_sr.py

```diff
@@ -0,0 +1,96 @@
+from conftest import GROUP_NAME
+from lib.common import wait_for
+from subprocess import CalledProcessError
+import pytest
+import time
+
+# Requirements:
+# - one XCP-ng host >= 8.2 with an additional unused disk for the SR
+# - access to XCP-ng RPM repository from the host
+
+class TestLinstorSRCreateDestroy:
+    vm = None
+
+    def test_create_sr_without_linstor(self, hosts, lvm_disks):
+        master = hosts[0]
+
+        # This test must be the first in the series in this module
+        assert not master.binary_exists('linstor'), \
+            "linstor must not be installed on the host at the beginning of the tests"
+        try:
+            # though it is expected to fail, result assigned to TestLinstorSR.sr for teardown in case it succeeds
+            sr = master.sr_create('linstor', 'LINSTOR-SR', {
+                'hosts': ','.join([host.hostname for host in hosts]),
+                'group-name': GROUP_NAME,
+                'redundancy': len(hosts),
+                'provisioning': 'thick'
+            }, shared=True)
+            try:
+                sr.destroy()
+            except Exception:
+                pass
+            assert False, "SR creation should not have succeeded!"
+        except CalledProcessError as e:
+            print("SR creation failed, as expected: {}".format(e))
+
+    def test_create_and_destroy_sr(self, linstor_sr, hosts_with_linstor, vm_ref):
+        # Create and destroy tested in the same test to leave the host as unchanged as possible
+        master = hosts_with_linstor[0]
+        # import a VM in order to detect vm import issues here rather than in the vm_on_xfs_fixture used in
+        # the next tests, because errors in fixtures break teardown
+        vm = master.import_vm_url(vm_ref, linstor_sr.uuid)
+        vm.destroy(verify=True)
```

Comment:
If I understand correctly, most of what the test is supposed to test is done in the `linstor_sr` fixture. On other tests, I purposely avoided using the `sr_xxx` fixture so that the test core stays in the test code (Pytest handles and displays fixture failures and failures from within tests differently) and the fixture stays available in the session scope for sharing between tests (basic tests + migration tests...).